### PR TITLE
Rename TuistConfig.swift to Config.swift and read it from Tuist/

### PR DESCRIPTION
**Pull request by pepibumur**
### Short description 📝
This is something that I've been pondering for a while and that I finally found some time to implement. 

Right now, the files under `Tuist/` and `TuistConfig.swift` act as a global elements. They are used by Tuist when it's run from any of the subdirectories. When we load the graph, we traverse up the directories hierarchy to find those. In case of the former, we read the `ProjectDescriptionHelpers` from it *(and probably other things in the future)*, and with the latter we configure the generation of projects.

Since both have the same nature, configure things globally within a repository that contains one or more projects, I think it's more intuitive to keep all the global elements together under the same directory. Since we are already using `Tuist/` as that global directory, I changed the logic in Tuist to read the file from `Tuist/Config.swift`. Moreover, since `Tuist/` already indicates that the files under it are associated to Tuist, there's no need to be redundant and include `Tuist....swift` in the name.

**Not a breaking change**
Please, note that this is not a breaking change. I implemented it in a way that it fallbacks to the current way of defining the configuration. I updated all the fixtures to follow the new convention and added a new fixture that has the configuration defined following the old convention, `app_with_old_config_name`.

As part of the PR, I updated the documentation and the init command.

### CODE REVIEW

File: Sources/TuistLoader/Loaders/GeneratorModelLoader.swift [hunk 1]
```diff
@@ -39,17 +49,35 @@ public class GeneratorModelLoader: GeneratorModelLoading {
...
+                let manifest = try manifestLoader.loadConfig(at: configPath.parentDirectory)
+                return try TuistCore.Config.from(manifest: manifest)
+            }
+        }
+
+        // We first try to load the deprecated file. If it doesn't exist, we load the new file name.
+        let fileNames = [Manifest.config]
+            .flatMap { [$0.deprecatedFileName, $0.fileName] }
+            .filter { $0 != nil }
+            .map { $0! }
```
Comment (ollieatkinson) [inline Sources/TuistLoader/Loaders/GeneratorModelLoader.swift]: could this be a `compactMap` ?

---

You could even write an extension so you don't need to specify `$0`

```swift
extension Array {
    
    @inlinable public func compact() rethrows -> [Element] {
        compactMap { $0 }
    }
    
}
```

File: Sources/TuistSupport/Utils/Functions.swift [hunk 2]
```diff
@@ -5,8 +5,7 @@
 ///
 /// - Parameter keyPath: A key path.
 /// - Returns: A getter function.
-public func get<Root, Value>(_ keyPath: KeyPath<Root, Value>) -> (Root) -> Value {
-    { root in root[keyPath: keyPath] }
+public func get<Root, Value>(_ keyPath: KeyPath<Root, Value>) -> (Root) -> Value { { root in root[keyPath: keyPath] }
```
Comment (ollieatkinson) [inline Sources/TuistSupport/Utils/Functions.swift]: this formatting looks odd, what version of `swiftformat` are you using?

File: Sources/TuistLoader/Loaders/ManifestLoader.swift [hunk 3]
```diff
@@ -134,12 +134,19 @@ public class ManifestLoader: ManifestLoading {
     // MARK: - Private
 
     private func loadManifest<T: Decodable>(_ manifest: Manifest, at path: AbsolutePath) throws -> T {
-        let manifestPath = path.appending(component: manifest.fileName)
-        guard FileHandler.shared.exists(manifestPath) else {
-            throw ManifestLoaderError.manifestNotFound(manifest, path)
+        var fileNames = [manifest.fileName]
+        if let deprecatedFileName = manifest.deprecatedFileName {
```
Comment (ollieatkinson) [inline Sources/TuistLoader/Loaders/ManifestLoader.swift]: Currently the manifest doesn't have a `deprecatedFileName` - was this added to support a future case?

Comment (pepibumur) [reply Sources/TuistLoader/Loaders/ManifestLoader.swift]: It's [defined here](https://github.com/tuist/tuist/pull/1083/files#diff-7c4da501bdfa517ff825e22a32bb2a19R11). It's intended to be used for this case. Once we drop the support for the current convention, we can remove that code.

File: Sources/TuistSupport/Utils/Functions.swift (hunk 2 above)
Comment (pepibumur) [reply Sources/TuistSupport/Utils/Functions.swift]: I'm using the latest, 0.44.4 🤔

**Comment by pepibumur** [comment]
I addressed your comments here @ollieatkinson

**Pull request merged**